step-automation-packages/step-automation-packages-schema/src/main/java/step/automation/packages/yaml/schema/YamlScheduleSchemaGenerator.java

```diff
@@ -55,10 +56,13 @@ public JsonObjectBuilder createScheduleDefs() throws JsonSchemaPreparationExcept
         res.add("type", "object");
 
         JsonObjectBuilder schedulerTaskProperties = jsonProvider.createObjectBuilder();
-        schemaHelper.extractPropertiesFromClass(this.jsonSchemaCreator, AutomationPackageSchedule.class, schedulerTaskProperties, "scheduler");
+
+        List<String> requiredProperties = new ArrayList<>();
+        schemaHelper.extractPropertiesFromClass(this.jsonSchemaCreator, AutomationPackageSchedule.class, schedulerTaskProperties, "scheduler", requiredProperties);
 
         res.add("properties", schedulerTaskProperties);
         res.add("additionalProperties", false);
+        schemaHelper.addRequiredProperties(requiredProperties, res);
```

Comment:
Couldn't we delegate the complete schema creation to the schema helper instead of having a part of the logic?

Reply:
@jeromecomte 
Good idea. I'll try to extract more reusable logic to the schema helper.

Reply:
@iegorov777 @david-stephan 
I've extracted the method to `schemaHelper` and reused it.